server/src/main/java/org/elasticsearch/search/aggregations/pipeline/movfn/MovingFunctions.java

```diff
@@ -0,0 +1,360 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.pipeline.movfn;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.function.ToDoubleFunction;
+
+/**
+ * Provides a collection of static utility methods that can be referenced from MovingFunction script contexts
+ */
+public class MovingFunctions {
+
+    /**
+     * Find the maximum value in a window of values.
+     * If all values are missing/null/NaN, the return value will be NaN
+     */
+    public static double windowMax(Collection<Double> values) {
+        return values.stream().max(Double::compareTo).orElse(Double.NaN);
+    }
+
+    /**
+     * Find the minimum value in a window of values
+     * If all values are missing/null/NaN, the return value will be NaN
+     */
+    public static double windowMin(Collection<Double> values) {
+        return values.stream().min(Double::compareTo).orElse(Double.NaN);
+    }
+
+    /**
+     * Find the sum of a window of values
+     * If all values are missing/null/NaN, the return value will be 0.0
+     */
+    public static double windowSum(Collection<Double> values) {
+        if (values.size() == 0) {
+            return 0.0;
+        }
+        return values.stream().mapToDouble(value -> {
+            if (value != null && value.isNaN() == false) {
+                return value;
+            }
+            return 0.0;
+        }).sum();
+    }
+
+    /**
+     * Calculate a simple unweighted (arithmetic) moving average.
+     *
+     * Only finite values are averaged.  NaN or null are ignored.
+     * If all values are missing/null/NaN, the return value will be NaN.
+     * The average is based on the count of non-null, non-NaN values.
+     */
+    public static double simpleMovAvg(Collection<Double> values) {
```

Comment:
Maybe just call this `unweightedAvg`? Is the `MovAvg` suffix really necessary given this is within `MovingFunctions`? Ditto for the methods below.